library/general/containerlimits/template.yaml

```diff
@@ -210,10 +226,21 @@ spec:
 
         general_violation[{"msg": msg, "field": field}] {
           container := input.review.object.spec[field][_]
+          not is_exempt(container)
           mem_orig := container.resources.limits.memory
           mem := canonify_mem(mem_orig)
           max_mem_orig := input.parameters.memory
           max_mem := canonify_mem(max_mem_orig)
           mem > max_mem
           msg := sprintf("container <%v> memory limit <%v> is higher than the maximum allowed of <%v>", [container.name, mem_orig, max_mem_orig])
         }
+      libs:
+        - |
+          package lib.exempt_container
+
+          is_exempt(container) {
+            exempt_image_prefixes := object.get(object.get(input, "parameters", {}), "exemptImagePrefixes", [])
+            img := container.image
+            prefix := exempt_image_prefixes[_]
+            startswith(img, prefix)
```

Comment:
Are there any ways this might be circumvented by playing tricks with Docker image reference parsing idiosyncrasies? e.g. `library/alpine` vs.`alpine`, or `foobar/alpine` vs. `foo.bar/alpine`?

Reply:
Is there some convention people use to make sure the prefix of the docker container is absolutely resolved (i.e. represents the host repository).

There may be some element of users needing to know how to do the right thing here.

Reply:
It looks like it'd be on users to specify a fully-qualified image name:

https://www.redhat.com/en/blog/be-careful-when-pulling-images-short-name

I'm not sure there is anything we can do to make it better?

Reply:
It might be worth documenting a warning that highlights the potential ambiguity and urges users to fully qualify their prefixes and images.

Reply:
That's a good idea, we can add it to the description.